lib/AST/Type.cpp

```diff
@@ -3583,6 +3583,14 @@ SubstitutionMap TypeBase::getMemberSubstitutionMap(
 
 Type TypeBase::getTypeOfMember(ModuleDecl *module, const ValueDecl *member,
                                Type memberType) {
+  if (hasError())
```

Comment:
I think this should be `is<ErrorType>()` instead. That matches what `getContextSubstitutions` is checking for, and it should be okay to substitute into (e.g.) a bound generic type with one of its generic arguments having an error.